### DISCUSSION TITLE
What to do about the default console font size?

### DISCUSSION THREAD

**Discussion by guidocella** [mpv-player/mpv#15044] [Polls]
https://github.com/mpv-player/mpv/pull/14088 and https://github.com/mpv-player/mpv/pull/14114 have been closed, but due to multiple users asking how to increase the console font size in discussions and IRC, they are worth revisiting.

**Comment by Dudemanguy** [comment]
I must be in the minority or something. I've never had a problem with the default console font size. It's about the same font size as I use for the rest of my stuff.

**Reply by Raidfire-SDR** [reply]
Colours are impossible to use for anyone with colour blindness they need to go monochrome by default, had 2 days of hell trying to see the new lua menu even on a 65" TV.

**Comment by hooke007** [comment]
I don't understand why would someone like `Scale the console with the window height` ?
If my terminal's font size will be increased by the window height, i must treat it as a serious bug.

**Reply by Dudemanguy** [reply]
Stats currently scales with the window height but to me it is completely unreadable with any SD video size like 480p at the default font size. I prefer console's behavior.

But in fairness stats has way more text on it so I don't have a better idea on how to handle it.

**Reply by guidocella** [reply]
To keep it readable on very large screens and for consistencty with OSD/OSC/stats/VLC/MPC-HC. I don't use it either, but we're making every script respect `--osd-scale-by-window` so you can configure this preference in one place.

**Reply by kasper93** [reply]
OSD is not a terminal though. Also note that scaling that is discussed would be similar to stats.lua, where basically we have constant number of lines displayed on screen and if you make it bigger presumably to watch from further away it will scale to that.

If you watch something on 13.3 inch laptop from 30cm, this is not the same as watching from 5m on 85 inch screen. Using variable font size make it more universal.

It is up to discussion, but, now that we use select.lua more, it is important to keep it readable in most scenarios, without requiring user to adjust it constantly.

**Reply by hooke007** [reply]
To be clear, I agree that adding an option to `respecting --osd-scale-by-window`. But if `Scale the console with the window height` to be the new default value, it would be strange for me.

**Reply by guidocella** [reply]
`--osd-scale-by-window` defaults to yes, so how is it any more strange than everything else in mpv already scaling by default?

**Reply by hooke007** [reply]
I always treat it as a terminal.

**Reply by guidocella** [reply]
Other applications that aren't terminals don't scale either, it's just video players.

**Reply by Dudemanguy** [reply]
I don't mind the scaling behavior if we put a hard minimum size somewhere.

**Reply by hooke007** [reply]
> Other applications that aren't terminals don't scale either, it's just video players.

I post the my confusion but it make me more confused by your views. Still cannot get the point why we have to change it.

**Reply by guidocella** [reply]
As I wrote in the opening post, because multiple users are asking how to increase the font and border size because it's too small.

**Reply by guidocella** [reply]
We can also change scale by window options to only scale upwards if the height is > 720 if preferred.

**Reply by Dudemanguy** [reply]
>We can also change scale by window options to only scale upwards if the height is > 720 if preferred.

I like that one. Seems like a sensible cutoff to me. The only problem I guess is stats but maybe we just make all stats pages scrollable too?

**Reply by guidocella** [reply]
Yeah I'm not aware of disadvantages of making it scrollable other than just overriding up and down keybindings. It already gets clipped for me sometimes even at 1080p and there's also an open issue about it https://github.com/mpv-player/mpv/issues/11544

**Comment by Xianglos** [comment]
> multiple users asking how to increase the console font size in discussions and IRC

I don't find the option I want.
If we passed it, how should we do if some complain how to make the size fixed? Reverting the commit again?
Why not just add a script option to follow `--osd-scale-by-window` and let the user decide instead?

**Reply by guidocella** [reply]
What? The user decides the value of `--osd-scale-by-window`. Another script option to behave differently can also be added.

**Comment by bt4ibwem8** [comment]
If i run mpv in the small window, size the playlist items is ok, but when i then go fullscreen, the text is very small. So scaling should be enabled by default, because this would expectable behavior for most users. I voted for the third option.

**Comment by Samillion** [comment]
I understand there is a lot of conflicting thoughts on this issue, there is one fact remaining that can be seen in reports, depending on screen/window resolution, the text becomes too small to read comfortably or at all.

Whether the fix is to scale it or increase font size or whichever, it definitely needs to be done. The logistics doesn't really matter.

To please everyone, do both and make "off" by default.

**Reply by guidocella** [reply]
You can already configure a bigger font size. This is about what the defaults should be.

**Reply by Samillion** [reply]
Yup, first thing I did a while ago. Scale would be my choice,

**Comment by Andarwinux** [comment]
Scaling is important for multiple displays at different DPI.

**Reply by guidocella** [reply]
console already factors `display-hidpi-scale`. The 3rd option is about also scaling with the window height.

**Comment by avih** [comment]
All text output in mpv scales with the window, e.g. OSD, the stats page, etc.

I don't see why the console should be different.

IMO it should be proportional to the OSD size, so that e.g. if someone prefers bigger text, then the console should also have bigger text (same for smaller).

I think a reasonable default size is 40% (of the default/configured OSD size). Or maybe the same size as in stats in general or maybe of page 4 specifically (keybindings).

**Reply by guidocella** [reply]
Agreed. This is also convenient if you use the same font in both places and change font often. ~I tried to do this in stats too in 51bd00c33a but text printed by `\\fsn` is way bigger than OSD text printed with `--osd-font-size=n` for some reason and couldn't figure out how to replicate the same size.~ this is complicated and depends on whether `show-text` or `osd-overlay` and `--osd-scale-by-window` are used

**Comment by stax76** [comment]
I've customized the console font size to use 18 instead of 16, not having the best pair of eyes.

**Comment by Raidfire-SDR** [comment]
I have MS with the optical neuritis to boot, when i look at a line of text i get a perfectly in focus mirror image both above and below the line im looking at with spacing matching the text line spacing so in effect for every line on the page i see 3 lines superimposed on each other, add to that almost total colour blindness. The vertical line issue being the main reason why I can't use the new subs menu, 3 words on top of each other in a colour that blends into the picture simply didn't work for me, out of the box the console was unusable.

On my first ventures with MPV a couple of years ago the first thing i fixed was the visibility issues in my main conf dpi scaling is off (in my conf i have MPV nailed down to a 65" sammy, i never use it on a standard monitor), osd-font-size = 60, in my /script-opts/console.conf i have font_size=20. this worked well for a couple of years, yesterday whilst discussing the new lua menu with @guidocella he suggested a font change, using the heavier sans-serif now makes the interface almost perfect, the old man can see it and his better equipped family can still tolerate using the interface as it's not too 'in your face'.

Just my 2 pennies worth on the debate.....

**Reply by Raidfire-SDR** [reply]
If fact I'd like to make an offer - I have the time to work on an official 'Limited Vision', accessibility script, we could use a batch file to push it into a users main conf if they want to use it.

Happy to own it if it get's the vote......

Also i ask myself - how well would the engine cope if we added the ability to change text size on the fly as long as we set some reasonable hard limits.

**Reply by Samillion** [reply]
If you do, please let me know. Official or not, I'll gladly link it in my mpv related repos.

**Reply by Raidfire-SDR** [reply]
Ok, I'll see what i can come up with in the week, been meaning to push out my full config again, plays nice with both sdr and hdr with metadata pass thru, spdif passthru and speaker configs on button activated profiles, auto vsync and scaling profiles for anything under 60fps or 2160p, osd on / off and stats menu available on the mouse (fwd, bck, middle) and a few gsl scripts on the keypad.

**Comment by guidocella** [comment]
Increased with #14114.
